otel/otel.go

```diff
@@ -0,0 +1,171 @@
+package otel
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	otelsdk "go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp"
+	"go.opentelemetry.io/otel/exporters/stdout/stdoutmetric"
+	"go.opentelemetry.io/otel/metric"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.26.0"
+)
+
+// Config defines the Otel Wrapper configuration for metrics
+type Config struct {
+	ServiceName               string
+	Debug                     bool
+	ExporterType              string
+	ExporterEndpoint          string
+	ExporterTimeout           string
+	ExporterCertificate       string
+	ExporterClientCertificate string
+	ExporterClientKey         string
+	CollectionInterval        time.Duration
+	MetricsReader             sdkmetric.Reader
+	MetricsExporter           sdkmetric.Exporter
+}
+
+// OtelWrapper encapsulates OpenTelemetry MetricProvider and utilities
+type OtelWrapper struct {
+	meterProvider *sdkmetric.MeterProvider
+	globalMeter   metric.Meter
+	config        Config
+}
+
+var (
+	globalWrapper *OtelWrapper
+
+	// singletonShutdown holds the shutdown function once the meter provider is initialized.
+	singletonShutdown func(context.Context) error
+
+	// once ensures that initialization happens only once.
+	globalOnce sync.Once
+
+	// initErr captures any error encountered during initialization.
+	initErr error
+)
+
+// InitSingleton initializes the MeterProvider as a singleton.
+// Subsequent calls to this function will return the same shutdown function and error.
+// The configuration is used only on the first call.
+func InitGlobalWrapper(ctx context.Context, cfg Config) (func(context.Context) error, error) {
+	globalOnce.Do(func() {
+		singletonShutdown, initErr = New(ctx, cfg)
+	})
+	return singletonShutdown, initErr
+}
```

Comment:
For me this function is not needed. This can be done inside the `New` construct directly. It will remove complexity.

Only `New` should be exported here i think